src/algorithms/searches/binarySearch/BinarySearchTemplated.java

```diff
@@ -0,0 +1,66 @@
+package src.algorithms.searches.binarySearch;
```

Comment:
for the templated version, can also construct ur own problem / source from tutorial qn (this would be best) or give a link to a leetcode BS qn like koko eating bananas